eng/Build-Native.cmd

```diff
@@ -174,154 +174,145 @@ if not exist "%__DotNetCli%" (
 
-    pushd "%__IntermediatesDir%"
-    call "%__ProjectDir%\eng\gen-buildsys-win.bat" "%__ProjectDir%" %__VSVersion% %__BuildArch% !__ExtraCmakeArgs!
+    pushd "%__CrossCompIntermediatesDir%"
+    call "%__ProjectDir%\eng\gen-buildsys-win.bat" "%__ProjectDir%" %__VSVersion% %__CrossArch% !__ExtraCmakeArgs!
     @if defined _echo @echo on
     popd
 
-:SkipConfigure
-    if defined __ConfigureOnly goto SkipNativeBuild
-
-    if not exist "%__IntermediatesDir%\install.vcxproj" (
-        echo %__MsgPrefix%Error: failed to generate native component build project!
+:SkipConfigureCrossBuild
+    if not exist "%__CrossCompIntermediatesDir%\install.vcxproj" (
+        echo %__MsgPrefix%Error: failed to generate cross-arch components build project!
         exit /b 1
     )
-    set __BuildLog="%__LogDir%\Native.Build.binlog"
+    if defined __ConfigureOnly goto SkipCrossCompBuild
+
+    set __BuildLog="%__LogDir%\Cross.Build.binlog"
 
     :: For some currently unknown reason, "%__DotNetCli% msbuild" fails because VCTargetsPath isn't defined.
-    msbuild.exe %__IntermediatesDir%\install.vcxproj /v:!__Verbosity! /bl:!__BuildLog! /p:Configuration=%__BuildType% /p:Platform=%__BuildArch% %__UnprocessedBuildArgs%
+    msbuild.exe %__CrossCompIntermediatesDir%\install.vcxproj /v:!__Verbosity! /bl:!__BuildLog! /p:Configuration=%__BuildType% /p:Platform=%__CrossArch% %__UnprocessedBuildArgs%
 
     if not !ERRORLEVEL! == 0 (
-        echo %__MsgPrefix%Error: native component build failed. Refer to the build log files for details:
+        echo %__MsgPrefix%Error: cross-arch components build failed. Refer to the build log files for details:
         echo     !__BuildLog!
         exit /b 1
     )
 
-:SkipNativeBuild
+:SkipCrossCompBuild
     rem } Scope environment changes end
     endlocal
 )
 
 REM =========================================================================================
 REM ===
-REM === Build Cross-Architecture Native Components (if applicable)
+REM === Build the native code
 REM ===
 REM =========================================================================================
 
-if /i "%__DoCrossArchBuild%"=="1" (
+if %__Build% EQU 1 (
     rem Scope environment changes start {
     setlocal
 
-    echo %__MsgPrefix%Commencing build of cross architecture native components for %__BuildOS%.%__BuildArch%.%__BuildType%
+    echo %__MsgPrefix%Commencing build of native components for %__BuildOS%.%__BuildArch%.%__BuildType%
 
-    :: Set the environment for the native build
     set __VCBuildArch=x86_amd64
-    if /i "%__CrossArch%" == "x86" ( set __VCBuildArch=x86 )
+    if /i "%__BuildArch%" == "x86" ( set __VCBuildArch=x86 )
+    if /i "%__BuildArch%" == "arm" (
+        set __VCBuildArch=x86_arm
+        :: Make CMake pick the highest installed version in the 10.0.* range
+        set ___SDKVersion="-DCMAKE_SYSTEM_VERSION=10.0"
+    )
+    if /i "%__BuildArch%" == "arm64" (
+        set __VCBuildArch=x86_arm64
```

Comment:
Are we going to compile for arm64 on an x86 machine on windows?

Reply:
This came over from coreclr's scripts.  I left it because someday maybe.